### PR TITLE
Remove lodash dependency

### PR DESCRIPTION
**Pull request by Tvrqvoise**
This removes the Lodash dependency to decrease the overall size of the dist file from 120K to 16K.

### CODE REVIEW

**Comment by Tvrqvoise** [comment]
@tyv Mind giving this a quick look?  This was kind of a conservative change -- an alternative solution would be to pass the files through Babel instead of bundling them with Webpack, which would also prevent the lodash modules from being included twice in projects.

**Comment by tyv** [comment]
@Tvrqvoise as far as I know with npm 3.x you will not bundle this files twice

**Comment by Tvrqvoise** [comment]
I'm actually using npm 3.x, but it's not really an npm problem.  It's more related to the files being bundled together and thus no longer able to be optimized by Webpack.

Here's the issue:

When we run Webpack in pick-react-known-props, we take our source files, add a bunch of lodash stuff on top of it, and bundle it into a single dist file.

Later, I import pick-react-known-prop into my project.  My project already depends on the same lodash files.  When I run Webpack on my project it sees the lodash files I depend upon and will try to dedup them -- but it doesn't see the lodash files inside pick-react-known-prop because they're "hidden" within the bundle.  Therefore we end up with the "real" lodash files as well as the ones bundled inside the dist file.

If you passed your project through Babel directly instead of Webpack, that would also fix the issue because then Webpack would see the actual imports being made by your source files and know how to handle them.  That being said, I don't know if you really need lodash for how small this module is anyway, and extra dependencies just mean extra headaches in my opinion.

**Comment by tyv** [comment]
@Tvrqvoise I agree, have few tiny comments regarding code

File: src/lib/pick-props.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+const isPlainObject = obj =>
```
Comment (tyv) [inline src/lib/pick-props.js]: is there a reason why it is not just `function isPlainObject(obj) {}`?

File: src/lib/pick-props.js [hunk 2]
```diff
@@ -0,0 +1,13 @@
...
+  typeof obj === 'object' && obj !== null && obj.constructor === Object && Object.prototype.toString.call(obj) === '[object Object]'
+
```
Comment (tyv) [inline src/lib/pick-props.js:2]: if we put a lodash function replacement, it seems reasonable if it will have own tests.

**Review by tyv — CHANGES_REQUESTED** [review]

File: src/lib/pick-props.js (hunk 2 above)
Comment (tyv) [reply src/lib/pick-props.js:2]: Ah, we have test with empty array. Think that is enough.

File: src/lib/pick-props.js (hunk 1 above)
Comment (Tvrqvoise) [reply src/lib/pick-props.js]: Not really -- changed it to a plain function.

File: src/lib/pick-props.js (hunk 2 above)
Comment (Tvrqvoise) [reply src/lib/pick-props.js:2]: Didn't see your second comment, so I ported over a bunch of the tests from lodash for `isPlainObject`.  There were some areas where they weren't quite the same (such as if the object contained a custom-defined `constructor` property), so it's probably a good thing in the end.

**Comment by Tvrqvoise** [comment]
@tyv Okay, updated the PR with requested changes.  Let me know if all looks good.

**Comment by tyv** [comment]
@Tvrqvoise thank you very much, looks fine. 
I am sorry didn't thought about this before, but if we do the copy of lodash's `isPlainObject` can you  put it in separate file in `./src/lib` folder with name `lodash.is-plain-object.js` please.

**Comment by Tvrqvoise** [comment]
@tyv Done, let me know if there's anything else or if you want me to squash the commit.

**Comment by tyv** [comment]
@Tvrqvoise thanks! thats fine, I will do the version today a bit later.

**Comment by tyv** [comment]
published

**Pull request merged**